### PR TITLE
fix(linter): C005 awk program assignment literals

### PR DESCRIPTION
**Pull request by ewhauser**
## Summary
- classify residual C005 zsh findings and fix the narrow awk-program assignment bucket outside the alias literal suffix work in #897
- exempt lower-case `awk_script` / `awk_program` assignment literals only when the text looks like an awk program with field references
- keep ordinary shell literals, uppercase compat fixtures, path assignments, and output text reporting as before

## Sampled C005 classifications
- `/tmp/zsh/ohmyzsh/lib/cli.zsh:814`: bug, an `awk_script` literal program assembled for later awk use; `$0` / `$1` and backtick text are awk/source text, not shell expansion mistakes
- `/tmp/zsh/holman-dotfiles/git/completion.zsh:4`: valid diagnostic, `completion='$(brew --prefix)/...'` does not evaluate before `test -f` / `source`
- `/tmp/zsh/dotfiler/example_install/02-shell-utils.zsh:144`: residual policy/literal-teaching text, intentionally printed install command, not handled in this PR
- `/tmp/zsh/ohmyzsh/lib/diagnostics.zsh:185`: residual literal output label, intentionally printed text, not handled in this PR

## Validation
- regression test failed before the matcher change
- `cargo fmt -- --check`
- `cargo test -p shuck-linter --lib single_quoted_literal::tests::`
- `cargo test -p shuck-linter`
- `make test-large-corpus SHUCK_LARGE_CORPUS_RULES=C005`
- targeted C005 shuck check on the sampled `/tmp/zsh` files shows `ohmyzsh/lib/cli.zsh:814` cleared while the other sampled residuals still report

### CODE REVIEW

File: crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs [hunk 1]
```diff
@@ -172,14 +172,43 @@ fn powerlevel10k_prompt_expansion_target(target: &str) -> bool {
     target.starts_with("POWERLEVEL9K_") && target.ends_with("_EXPANSION")
 }
 
+fn assignment_target_program_text_is_exempt(target: &str, text: &str) -> bool {
+    assignment_target_looks_like_awk_program(target) && text_looks_like_awk_program(text)
+}
+
+fn assignment_target_looks_like_awk_program(target: &str) -> bool {
+    target.contains("awk") && (target.contains("script") || target.contains("program"))
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs]: Fixed in `3c0fa02bc`: the matcher now tokenizes assignment targets and requires exact `awk` plus `script`/`program` tokens, with regressions for `hawk_program`, `awkward_script`, and uppercase `AWK_SCRIPT`. Re-ran focused C005 tests, full `shuck-linter`, targeted sampled-file checks, and `make test-large-corpus SHUCK_LARGE_CORPUS_RULES=C005`.

File: crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs [hunk 2]
```diff
@@ -172,14 +172,49 @@ fn powerlevel10k_prompt_expansion_target(target: &str) -> bool {
...
+}
+
+fn assignment_target_looks_like_awk_program(target: &str) -> bool {
+    let mut has_awk_token = false;
+    let mut has_program_token = false;
+    for token in target.split(|character: char| !character.is_ascii_alphanumeric()) {
+        has_awk_token |= token == "awk";
+        has_program_token |= matches!(token, "script" | "program");
+    }
+    has_awk_token && has_program_token
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs]: Fixed in `2ce798bc3`: target matching now requires adjacent `awk` followed by `script` or `program` tokens, with regressions for `script_awk` and `program_awk` in addition to the earlier boundary cases. Re-ran focused tests, full `shuck-linter`, sampled-file checks, and the C005 large-corpus gate.

File: crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs [hunk 3]
```diff
@@ -172,14 +172,48 @@ fn powerlevel10k_prompt_expansion_target(target: &str) -> bool {
...
+            || text.contains("gsub(")
+            || text.contains("BEGIN")
+            || text.contains("END"))
+}
+
+fn contains_awk_field_reference(text: &str) -> bool {
+    let bytes = text.as_bytes();
+    let mut index = 0usize;
+    while index + 1 < bytes.len() {
+        if bytes[index] == b'$' && bytes[index + 1].is_ascii_digit() {
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs:227]: Fixed in `81af37c94`: the awk field-reference recognizer now also covers `$NF` and dynamic field expressions like `$(NF - 1)` while still not treating arbitrary `$HOME` as awk syntax. Re-ran focused C005 tests, full `shuck-linter`, sampled-file checks, and `make test-large-corpus SHUCK_LARGE_CORPUS_RULES=C005`.

File: crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs [hunk 4]
```diff
@@ -172,14 +172,52 @@ fn powerlevel10k_prompt_expansion_target(target: &str) -> bool {
...
 }
 
+fn assignment_target_program_text_is_exempt(target: &str, text: &str) -> bool {
+    assignment_target_looks_like_awk_program(target) && text_looks_like_awk_program(text)
+}
+
+fn assignment_target_looks_like_awk_program(target: &str) -> bool {
+    let tokens = target
+        .split(|character: char| !character.is_ascii_alphanumeric())
+        .collect::<Vec<_>>();
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs:183]: Fixed in `72108c319`: empty target-name tokens are filtered before matching adjacent `awk` -> `script`/`program`, with regressions for `awk__script` and `my_awk__program`. Re-ran focused tests, full `shuck-linter`, sampled-file checks, and the C005 large-corpus gate.

File: crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs [hunk 5]
```diff
@@ -172,14 +172,53 @@ fn powerlevel10k_prompt_expansion_target(target: &str) -> bool {
...
+}
+
+fn contains_awk_field_reference(text: &str) -> bool {
+    let bytes = text.as_bytes();
+    let mut index = 0usize;
+    while index + 1 < bytes.len() {
+        if bytes[index] == b'$' && bytes[index + 1].is_ascii_digit() {
+            return true;
+        }
+        if bytes[index] == b'$' && (bytes[index + 1] == b'(' || text[index + 1..].starts_with("NF"))
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs]: Fixed in `28c658183`: dynamic awk field matching now requires `$(` followed by `NF` or a digit after optional whitespace, so `$(NF - 1)` and `$( 2 )` are exempt but `$(printenv HOME)` is not. Re-ran focused tests, full `shuck-linter`, sampled-file checks, and the C005 large-corpus gate.

File: crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs [hunk 6]
```diff
@@ -172,14 +172,68 @@ fn powerlevel10k_prompt_expansion_target(target: &str) -> bool {
...
+    tokens
+        .windows(2)
+        .any(|window| window[0] == "awk" && matches!(window[1], "script" | "program"))
+}
+
+fn text_looks_like_awk_program(text: &str) -> bool {
+    contains_awk_field_reference(text)
+        && (text.contains("print")
+            || text.contains("sub(")
+            || text.contains("gsub(")
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-linter/src/rules/correctness/single_quoted_literal.rs]: Fixed in `2c3687825`: awk-program action detection now requires `print`, `BEGIN`, and `END` as bounded words and `sub`/`gsub` as function calls, so shell-like text such as `echo $1 printenv HOME` still reports C005. Re-ran focused C005 tests, full `shuck-linter`, sampled-file checks, and `make test-large-corpus SHUCK_LARGE_CORPUS_RULES=C005`.
